Libs/MRML/Core/vtkMRMLModelDisplayNode.h

```diff
@@ -73,23 +90,48 @@ class VTK_MRML_EXPORT vtkMRMLModelDisplayNode : public vtkMRMLDisplayNode
   /// \sa SetActiveScalarName()
   virtual void SetActiveAttributeLocation(int location);
 
-  /// Update the pipeline based on this node attributes
-  virtual void UpdatePolyDataPipeline();
+  /// Reimplemented to update scalar range accordingly
+  /// \sa SetActiveScalarName()
+  virtual void SetScalarRangeFlag(int flag);
+
+  /// Set whether to threshold the model display node.
+  /// \sa ThresholdEnabled, GetThresholdEnabled()
+  vtkSetMacro(ThresholdEnabled,bool);
+  vtkBooleanMacro(ThresholdEnabled,bool);
+
+  /// Get whether to threshold the model display node.
+  /// \sa ThresholdEnabled, SetThresholdEnabled(), ThresholdEnabledOn(),
+  /// ThresholdEnabledOff()
+  vtkGetMacro(ThresholdEnabled,bool);
+
+  /// Set the threshold range of the model display node.
+  /// \sa GetThresholdMin(), GetTresholdMax()
+  void SetThresholdRange(double min, double max);
+  void SetThresholdRange(double range[2]);
+
+  /// Get the threshold range of the model display node.
+  /// \sa SetThresholdRange()
+  void GetThresholdRange(double range[2]);
+  double GetThresholdMin();
+  double GetThresholdMax();
 
 protected:
   vtkMRMLModelDisplayNode();
   ~vtkMRMLModelDisplayNode();
   vtkMRMLModelDisplayNode(const vtkMRMLModelDisplayNode&);
   void operator=(const vtkMRMLModelDisplayNode&);
 
-  virtual void ProcessMRMLEvents(vtkObject *caller,
```

Comment:
> Updating the Assignattribute filter should only happen when ActiveScalarName or ActiveScalarLocation are updated.

Usually if you set the same value for a parameter in a VTK filter then the filter detects that and does not do any actual update. So, probably you don't need to check if ActiveScalarName or ActiveScalarLocation is changed.